polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -168,13 +179,56 @@ pub fn worker_entrypoint(
 					Arc::clone(&condvar),
 					WaitOutcome::TimedOut,
 				)?;
+
+				let fd = stream.as_raw_fd();
+				unsafe {
```

Comment:
nit: Is it possible to extract this `unsafe` block into a helper function? Or maybe just the `Box`? This low-level technical code is an implementation detail and can distract from what this function is doing at a high level.

Reply:
You mean just move it out of the main loop? Sure.

The whole call to `.start_tracking()` is unsafe as its arguments (evaluated before the actual call) must uphold the safety contract, so it cannot be just `Box`.